sample-code/src/ai-api/deployment-api.ts

```diff
@@ -0,0 +1,99 @@
+import { DeploymentApi } from '@sap-ai-sdk/ai-api';
+import type {
+  AiDeploymentCreationResponse,
+  AiDeploymentDeletionResponse,
+  AiDeploymentList,
+  AiDeploymentModificationResponse,
+  AiDeploymentResponseWithDetails
+} from '@sap-ai-sdk/ai-api';
+
+/**
+ * Get all deployments.
+ * @param resourceGroup - AI-Resource-Group where the resources are available.
+ * @param status - Optional parameter to filter deployments by status.
+ * @returns All deployments.
+ */
+export async function getDeployments(
+  resourceGroup: string,
+  status?:
+    | 'PENDING'
+    | 'RUNNING'
+    | 'COMPLETED'
+    | 'DEAD'
+    | 'STOPPING'
+    | 'STOPPED'
+    | 'UNKNOWN'
+): Promise<AiDeploymentList> {
+  // check for optional query parameters.
+  const queryParams = status ? { status } : {};
+  return DeploymentApi.deploymentQuery(queryParams, {
+    'AI-Resource-Group': resourceGroup
+  }).execute();
+}
+
+/**
+ * Get information about specific deployment.
+ * @param deploymentId - ID of the specific deployment.
+ * @param resourceGroup - AI-Resource-Group where the resources are available.
+ * @returns Details for deplyoment with deploymentId.
+ */
+export async function getDeployment(
```

Comment:
[pp] Also, I would really just put the functions here that are exposed through the `server.ts` file. Otherwise, when someone tries the sample it will be unused code from their point of view.